src/main.cpp

```diff
@@ -2126,9 +2126,12 @@ int64_t GetProofOfStakeReward(uint64_t nCoinAge, int64_t nFees, std::string cpid
     else
     {
             // Research Age Subsidy - PROD
-            int64_t nBoinc = ComputeResearchAccrual(nTime, cpid, operation, pindexLast, VerifyingBlock, VerificationPhase, dAccrualAge, dMagnitudeUnit, AvgMagnitude);
+            int64_t nBoinc = 0;
             int64_t nInterest = 0;
 
+            if (IsResearcher(cpid))
+                nBoinc = ComputeResearchAccrual(nTime, cpid, operation, pindexLast, VerifyingBlock, VerificationPhase, dAccrualAge, dMagnitudeUnit, AvgMagnitude);
```

Comment:
Please move this conditional into the `ComputeResearchAccrual` function, so the calling protocol will remain consistent also in other places.

Reply:
alright i'll move it to the top of ComputeResearchAccrual